### ISSUE TITLE
cache_get() requires value for second argument

### ISSUE THREAD

**Issue by cellear** [backdrop-ops/backdropcms.org#52]
I'm attempting to port the Honeypot module to backdrop, and out of the box I'm seeing this message when I go to the configuration screen:

>  Warning: Missing argument 2 for cache_get(), called in [...]/modules/honeypot/honeypot.module on line 141 and defined in cache_get() (line 59 of [...]/core/includes/cache.inc).
>  Notice: Undefined variable: bin in cache_get() (line 61 of [...]/core/includes/cache.inc).

I could work around this in the module by supplying an argument, but this strikes me as an API change, which I know we're trying to avoid.  I looked for a change record mentioning cache_get() but didn't find one.  Comparing D7 to Backdrop code, I can see that Drupal's version:

```php
function cache_get($cid, $bin = 'cache') {
  return _cache_get_object($bin)->get($cid);
}
```
in Backdrop has changed to this:

```php
function cache_get($cid, $bin) {
  return cache($bin)->get($cid);
}
```
It looks like the signature has changed.  Is there a reason for that?  If so, I believe we need a change record.  If not, perhaps we should change it back?

(Forgive me if this is an old issue, or if I'm missing something obvious, but I couldn't find mention of it anywhere.)

**Comment by docwilmot** [comment]
Yep, that needs a change record; its here https://github.com/backdrop/backdrop-issues/issues/446, but I havent documented it yet because was waiting on https://github.com/backdrop/backdrop-issues/issues/674 to get all 'cache_' issues all done up together. But in fact I would think this should be documented ASAP, and I'll do soon.

**Comment by cellear** [comment]
Btw, the "something obvious" that I was missing was that I was searching the wrong issue queue for "cache_get"  -- here, in backdrop-ops, instead of https://github.com/backdrop/backdrop-issues/.  Eek!  This issue relates to backdrop core code, and perhaps to the API site, but not really here, I don't think.  Should we close it? I can't tell if #446 covers it or not.  (It looks to me like if it did, I shouldn't be seeing the error anymore, because didn't PR 616 fix it?)

**Comment by docwilmot** [comment]
I think the change record will say that cache functions have changed; you'll still get errors because the parameters are different. The new cache API is still in Backdrop as well and you can use that instead of the "new" procedural wrappers. I think it would be safe to close this.

**Comment by quicksketch** [comment]
Let's restore the missing second argument to `cache_get()`. I didn't realize it was missing when I restored the procedural wrappers in https://github.com/backdrop/backdrop-issues/issues/446.

I filed a new issue at https://github.com/backdrop/backdrop-issues/issues/783 in the main issue queue. Let's continue there.
